extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/LinksProcessor.java

```diff
@@ -0,0 +1,90 @@
+import io.quarkus.resteasy.reactive.links.runtime.LinksProviderRecorder;
+import io.quarkus.resteasy.reactive.links.runtime.PathParameterInfo;
+import io.quarkus.resteasy.reactive.links.runtime.RestLinksProviderProducer;
+import io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveDeploymentInfoBuildItem;
+import io.quarkus.resteasy.reactive.spi.CustomContainerResponseFilterBuildItem;
+
+final class LinksProcessor {
+
+    @BuildStep
+    void feature(BuildProducer<FeatureBuildItem> feature) {
+        // TODO move to the main features definition
+        feature.produce(new FeatureBuildItem("resteasy-reactive-links"));
+    }
+
+    @BuildStep
+    LinksContextBuildItem createLinksContext(CombinedIndexBuildItem indexBuildItem,
+            ResteasyReactiveDeploymentInfoBuildItem deploymentInfoBuildItem) {
+        LinksContextFactory linksContextFactory = new LinksContextFactory(indexBuildItem.getIndex());
+        List<ResourceClass> resourceClasses = deploymentInfoBuildItem.getDeploymentInfo().getResourceClasses();
+
+        return new LinksContextBuildItem(linksContextFactory.getLinksContext(resourceClasses));
+    }
+
+    @BuildStep
+    List<BytecodeTransformerBuildItem> generateGetters(CombinedIndexBuildItem indexBuildItem,
+            LinksContextBuildItem linksContextBuildItem) {
+        IndexView index = indexBuildItem.getIndex();
+        List<BytecodeTransformerBuildItem> bytecodeTransformerBuildItems = new LinkedList<>();
+        Map<String, Set<PathParameterInfo>> pathParameters = new HashMap<>();
+
+        // Group all path parameters by entity type, to avoid duplicate getters generation
+        for (List<LinkInfo> linkInfos : linksContextBuildItem.getLinksContext().getMappedLinks().values()) {
+            for (LinkInfo linkInfo : linkInfos) {
+                if (!pathParameters.containsKey(linkInfo.getEntityType())) {
+                    pathParameters.put(linkInfo.getEntityType(), new HashSet<>());
+                }
+                pathParameters.get(linkInfo.getEntityType()).addAll(linkInfo.getPathParametersInfo());
+            }
+        }
+
+        for (Map.Entry<String, Set<PathParameterInfo>> entry : pathParameters.entrySet()) {
+            bytecodeTransformerBuildItems.add(new BytecodeTransformerBuildItem(entry.getKey(),
+                    PathParameterValueGettersGenerator.getVisitorFunction(index, entry.getKey(), entry.getValue())));
+        }
+
+        return bytecodeTransformerBuildItems;
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem registerRestLinksProviderProducer() {
+        return AdditionalBeanBuildItem.unremovableOf(RestLinksProviderProducer.class);
+    }
+
+    @BuildStep
+    CustomContainerResponseFilterBuildItem registerRestLinksResponseFilter() {
+        return new CustomContainerResponseFilterBuildItem(RestLinksResponseFilter.class.getName());
+    }
+
+    @Record(STATIC_INIT) // TODO this doesn't seem to work on reload
```

Comment:
What doesn't work?

Reply:
It doesn't seem to reload the LinksContext. I've just marked this one to look into once we've agreed on the API.